pombola/south_africa/views/attendance.py

```diff
@@ -53,20 +54,30 @@ def filter_attendance(self, annual_attendance, party, position):
             attendance_summary = [ma for ma in attendance_summary if
                 ma['member']['party_name'] == party]
 
-        if position == 'ministers':
-            active_ministers = Position.objects.filter(
+        year_end_date = datetime.datetime.strptime(
+            annual_attendance['end_date'], "%Y-%m-%d").date()
+        active_ministers = Position.objects.filter(
                 Q(title__slug__startswith='minister') |
                 Q(title__slug__startswith='deputy-minister')
-            ).overlapping_dates(
-                annual_attendance['start_date'],
-                annual_attendance['end_date'])
+            ).currently_active(
+                year_end_date)
```

Comment:
I've merged this anyway, because it sounds like it's time-sensitive and you might not be able to look at feedback tomorrow. But anyway, you could simplify this a little bit by doing:

```python
active_minister_slugs = set(Position.objects.filter(
    Q(title__slug__startswith='minister') |
￼    Q(title__slug__startswith='deputy-minister')).currently_active(year_end_date) \
        .values_list('person__slug', flat=True))
```